tests/internal/fstore.c

```diff
@@ -64,7 +64,7 @@ void cb_all()
     ret = stat(FSF_STORE_PATH "/abc/example.txt", &st_data);
     TEST_CHECK(ret == 0);
 
-    ret = flb_fstore_file_append(fsf, "fluent-bit\n", 11);
+    ret = flb_fstore_file_append(fs, fsf, "fluent-bit\n", 11);
     TEST_CHECK(ret == 0);
 
```

Comment:
Please extend the fstore.c unit test to try to reproduce the issue you were facing initially

Reply:
This sound good. Will add.

Reply:
Added a test!
Get the following errors without the patch: 
```
[2022/02/11 04:47:22] [error] [fstore] [cio file] file is not mmap()ed: abc:files-round_2-65.txt
[2022/02/11 04:47:22] [error] [fstore] could not write data to file files-round_2-65.txt
  fstore.c:172: Check ret == 0... failed
  fstore.c:177: Check memcmp(out_buf, "fluent-bit-3\n", 13) == 0... failed
  fstore.c:178: Check out_size == 13... failed
  fstore.c:169: Check ret == 0... failed
[2022/02/11 04:47:22] [error] [fstore] [cio file] file is not mmap()ed: abc:files-round_2-66.txt
[2022/02/11 04:47:22] [error] [fstore] could not write data to file files-round_2-66.txt
  fstore.c:172: Check ret == 0... failed
  fstore.c:177: Check memcmp(out_buf, "fluent-bit-3\n", 13) == 0... failed
  fstore.c:178: Check out_size == 13... failed
  fstore.c:169: Check ret == 0... failed
[2022/02/11 04:47:22] [error] [fstore] [cio file] file is not mmap()ed: abc:files-round_2-67.txt
[2022/02/11 04:47:22] [error] [fstore] could not write data to file files-round_2-67.txt
...
  fstore.c:193: Check out_size == 11+13... failed
FAILED: 1 of 2 unit tests has failed.
```

With the patch, the following is output showing success:
```
[ec2-user@ip-172-31-30-69 bin]$ /home/ec2-user/fala-forks/fluent-bit/build/bin/flb-it-fstore 
Test all...                                     [2022/02/11 04:50:17] [ info] [fstore] created root path /tmp/flb-fstore
===== FSTORE DUMP =====
- stream: abc
          abc/example.txt (up)

[ OK ]
Test many_files...                              [2022/02/11 04:50:17] [ info] [fstore] created root path /tmp/flb-fstore
[2022/02/11 04:50:17] [ info] [fstore] created root path /tmp/flb-fstore
[2022/02/11 04:50:17] [ info] [fstore] created root path /tmp/flb-fstore
[2022/02/11 04:50:17] [ info] [fstore] created root path /tmp/flb-fstore
[ OK ]
SUCCESS: All unit tests have passed.
```